pokemon-app/src/components/EnemyPokemon.js

```diff
@@ -70,13 +72,25 @@ const EnemyPokemon = ({ enemyPokemon, hasEnemy }) => {
               className={`enemy-pokemon__moves ${
                 enemyStatsOnTop ? "" : "on-top"
               }`}
+              ref={scrollRef}
+              onScroll={onScroll}
             >
               <ol className="enemy-pokemon__moves-list">
                 {!enemyPokemon?.moves
                   ? null
                   : enemyPokemon?.moves.map((moveData, moveIndex) => (
                       <li key={moveIndex}>{moveData.move.name}</li>
                     ))}
+
+                {!enemyPokemon?.moves ? null : !(scrollTop > 0) ? null : (
```

Comment:
This is interesting logic here. I'm not entirely sure I understand it. I would imagine that it is working well though.

Reply:
Hmmm.  Good point.  It might be a good idea to pull the enemy pokemon moves list block out into a function and document it for better readability.

Reply:
Basically, the first conditional checks if there are any `moves`, and if not then don't render anything. The second checks if the user has scrolled away from the top. To know that is to check if it is greater than 0 since that represents the user is still at the top. Anything greater than 0 means that they scrolled towards the bottom.